system-addon/lib/TopSitesFeed.jsm

```diff
@@ -65,9 +69,15 @@ this.TopSitesFeed = class TopSitesFeed {
       }
     }
 
-    // Now, get a screenshot for every item
+    // Now, get a tippy top icon or screenshot for every item
     for (let link of links) {
       if (!link) { continue; }
+
+      // Check for tippy top icon.
+      link = this._tippyTopProvider.processSite(link);
```

Comment:
Well, Provider doesn't have the usual dispatch, but it async init returns a promise that TopSitesFeed could use to have refresh wait until Provider is `init`ed . E.g.,
```
TopSitesFeed {
constructor() {
  const provider = new TippyTopProvider();
  this._initedProvider = provider.init(); // init needs to return the provider
}
async refresh() {
  …
  (await this._initedProvider).processSite(link);
```
This can be separately handled as a followup as I expect the tests would need to change a bit.